File: examples/benchmark.ts
// benchmark.ts — Recursive Fibonacci
// tscc compiles this with integer narrowing, matching native Rust performance.
// Compile: tscc build examples/benchmark.ts
// Run:     tscc run examples/benchmark.ts --benchmark

function fib(n: number): number {
    if (n <= 1) {
        return n
    }
    return fib(n - 1) + fib(n - 2)
}

console.log(fib(50))

File: examples/benchmark.rs
fn fib(n: f64) -> f64 {
    if n <= 1.0 {
        return n;
    }
    fib(n - 1.0) + fib(n - 2.0)
}

fn main() {
    let result = fib(40.0);
    println!("{}", result);
}
